src/auth.ts

```diff
@@ -158,15 +151,66 @@ async function authorizeWithoutLocalhost() {
  * @param {boolean} options.localhost authorize without http server.
  * @param {string} options.creds location of credentials file.
  */
-export function login(options: { localhost: boolean, creds: string}) {
-  DOTFILE.RC.read().then((rc: ClaspSettings) => {
-    console.warn(ERROR.LOGGED_IN);
-  }).catch(async (err: string) => {
-    DOTFILE.RC_LOCAL.read().then((rc: ClaspSettings) => {
-      console.warn(ERROR.LOGGED_IN);
-    }).catch(async (err: string) => {
-      await checkIfOnline();
-      authorize(options.localhost, options.creds);
-    });
-  });
+export async function login(options: { localhost: boolean, creds: string }) {
+  if ((options.creds && localOathSettingsExist()) ||
+      (!options.creds && defaultOathSettingsExist())) {
+    logError(null, ERROR.LOGGED_IN);
+  }
+  await checkIfOnline();
+  let ownCreds = false;
+  try {
+    const credentials = JSON.parse(fs.readFileSync(options.creds, 'utf8'));
+    if (credentials && credentials.installed && credentials.installed.client_id
+      && credentials.installed.client_secret) {
+        oauth2ClientSettings.clientId = credentials.installed.client_id;
+        oauth2ClientSettings.clientSecret = credentials.installed.client_secret;
+        ownCreds = true;
+        console.log(LOG.CREDENTIALS_FOUND);
+    } else {
+      logError(null, ERROR.BAD_CREDENTIALS_FILE);
+    }
+  } catch(err) {
+    if (err.code === 'ENOENT') {
+      logError(null, ERROR.NO_CREDENTIALS);
+    }
+    console.log(LOG.DEFAULT_CREDENTIALS);
+  }
+  await authorize(options.localhost, ownCreds);
+  process.exit(0); // gracefully exit after successful login
+}
+
+/**
+ * Set global OAuth client credentails from rc, save new if access token refreshed.
+ * @param {ClaspSettings} rc OAuth client settings from rc file.
+ */
+async function setOauthCredentials(rc: ClaspSettings) {
+  try {
+    await checkIfOnline();
+    if (isLocalCreds(rc)) {
+      // set global OAuth client settings for authorize
```

Comment:
I'll have to look if we can use the Node client library to not do these hacks.

Reply:
Currently no setter for client ID and secret after `OAuth2Client` constructor. May have to initialize API clients in `commands.ts` methods.